tuxemon/client.py

```diff
@@ -181,24 +189,33 @@ def draw_event_debug(self):
 
             yy = y
 
-    def process_events(self, events):
-        """Process all events for this frame.
+    def process_events(
+        self,
+        events: Iterable[pg.event.Event],
+    ) -> Generator[pg.event.Event, None, None]:
+        """
+        Process all events for this frame.
 
         Events are first sent to the active state.
         States can choose to keep the events or return them.
-        If they are kept, no other state nor the event engine will get that event.
+        If they are kept, no other state nor the event engine will get that
+        event.
         If they are returned, they will be passed to the next state.
         Kept or returned, the state may process it.
-        Eventually, if all states have returned the event, it will go to the event engine.
+        Eventually, if all states have returned the event, it will go to the
+        event engine.
         The event engine also can keep or return the event.
         All unused events will be added to Client.key_events each frame.
         Conditions in the the event system can then check that list.
 
-        States can "keep" events by simply returning None from State.process_event
+        States can "keep" events by simply returning None from
+        State.process_event
 
-        :param events: Sequence of pg events
-        :returns: Iterator of game events
-        :rtype: collections.Iterable[pg.event.Event]
+        Parameters:
+            events: Sequence of Pygame events]
```

Comment:
trailing `]`?

Reply:
In fact, I have a question here: should the type of the events be `PlayerInput` instead? It seems that is the case in `State`.

Reply:
Yeah, good point.  We are no longer using pygame events in the states.  The player events abstract that away so they do not need to know if it was a keyboard, joystick, or anything different.